### PR TITLE
Update cluster running status when provisioning has completed

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by csrwng** [comment]
@dgoodwin ptal

File: pkg/controller/machineset/machineset_controller.go [hunk 1]
```diff
@@ -253,6 +262,58 @@ func (c *Controller) handleErr(err error, key interface{}) {
...
+		}
+		machineSet, err := c.getMachineSet(key)
+		if err != nil {
+			return err
+		}
+		cluster, err := controller.ClusterForMachineSet(machineSet, c.clustersLister)
+		if err != nil {
+			return err
+		}
+		if cluster.Status.Running || !cluster.Status.Provisioned {
```
Comment (dgoodwin) [inline pkg/controller/machineset/machineset_controller.go]: Should we rename this from Running to Ready?

File: pkg/controller/machineset/machineset_controller.go [hunk 2]
```diff
@@ -253,6 +262,58 @@ func (c *Controller) handleErr(err error, key interface{}) {
...
+		return nil, err
+	}
+	if len(namespace) == 0 || len(name) == 0 {
+		return nil, fmt.Errorf("invalid key %q: either namespace or name is missing", key)
+	}
+	return c.machineSetsLister.MachineSets(namespace).Get(name)
+}
+
+func (c *Controller) checkClusterRunning(syncFunc func(string) error) func(string) error {
+	return func(key string) error {
```
Comment (dgoodwin) [inline pkg/controller/machineset/machineset_controller.go]: Might be helpful to have some debug logs sprinkled through here for some easier visibility as to why a cluster isn't yet marked ready.

File: pkg/controller/machineset/machineset_controller.go [hunk 3]
```diff
@@ -253,6 +262,58 @@ func (c *Controller) handleErr(err error, key interface{}) {
...
+		machineSets, err := controller.MachineSetsForCluster(cluster, c.machineSetsLister)
+		if err != nil {
+			return nil
+		}
+		if len(machineSets) != len(cluster.Spec.MachineSets) {
+			return nil
+		}
+		for _, ms := range machineSets {
+			if ms.Spec.NodeType == clusteroperator.NodeTypeMaster {
+				if !ms.Status.Provisioned || !ms.Status.Installed || !ms.Status.ComponentsInstalled || !ms.Status.NodeConfigInstalled {
```
Comment (dgoodwin) [inline pkg/controller/machineset/machineset_controller.go]: Should we be checking Accepted here as well for masters?

Comment (csrwng) [reply pkg/controller/machineset/machineset_controller.go]: masters are automatically joined to the cluster and don't need to run a separate accept step

**Comment by csrwng** [comment]
@dgoodwin renamed Running to Ready in cluster status and added debug logging to the function that checks whether the cluster is ready.

**Pull request merged**